src/components/views/elements/Field.tsx

```diff
@@ -257,17 +273,22 @@ export default class Field extends React.PureComponent<PropShapes, IState> {
         }
 
         const hasValidationFlag = forceValidity !== null && forceValidity !== undefined;
-        const fieldClasses = classNames("mx_Field", `mx_Field_${this.props.element}`, className, {
-            // If we have a prefix element, leave the label always at the top left and
-            // don't animate it, as it looks a bit clunky and would add complexity to do
-            // properly.
-            mx_Field_labelAlwaysTopLeft: prefixComponent || usePlaceholderAsHint,
-            mx_Field_placeholderIsHint: usePlaceholderAsHint,
-            mx_Field_valid: hasValidationFlag ? forceValidity : onValidate && this.state.valid === true,
-            mx_Field_invalid: hasValidationFlag
-                ? !forceValidity
-                : onValidate && this.state.valid === false,
-        });
+        const fieldClasses = classNames(
+            "mx_Field",
+            `mx_Field_${typeof this.props.element === "string" ? this.props.element : "input"}`,
```

Comment:
overriding `element` to be a string or component feels weird as it kinda implies that we can use a component name for `element`, which won't work. Can we just define it as `componentClass` in the props instead?

Reply:
The overloads already force the `element` to be `"input"`, `"select"`, `"textarea"` or a `ComponentClass`. If you try to enter a random string, it says `Type '"MyCustomThing"' is not assignable to type 'ComponentClass'.`

> Can we just define it as `componentClass` in the props instead?

Updated to use `props.componentClass` though ⏩